### PR TITLE
mappings traversal bug fix

### PR DESCRIPTION
**Pull request by petardz**
*Issue #, if available:*

*Description of changes:*

Fixed few bugs with index mappings traversal which resulted in monitors not working.

*CheckList:*
[ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by getsaurabh02 — APPROVED** [review]

File: alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt [hunk 1]
```diff
@@ -100,19 +100,34 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
         return clusterState.routingTable.hasIndex(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
     }
 
-    /**
-     * From given index mapping node, extracts fieldName -> fieldProperties pair
-     */
-    fun extractField(node: MutableMap<String, Any>, currentPath: String): Pair<String, MutableMap<String, Any>> {
+    fun traverseMappingsAndUpdate(
```
Comment (lezzago) [inline alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt:103]: Can you add comments in the different logic flows of this function? It is kind of hard to follow what it is doing.

Comment (petardz) [reply alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt:112]: done

**Review by getsaurabh02 — APPROVED** [review]

**Review by lezzago — APPROVED** [review]
